packages/hash/frontend/src/pages/[account-slug]/types/entity-type/multiple-values-cell.tsx

```diff
@@ -0,0 +1,162 @@
+import { TextField } from "@hashintel/hash-design-system/text-field";
+import {
+  Box,
+  Checkbox,
+  ClickAwayListener,
+  Collapse,
+  Fade,
+  Popper,
+  TableCell,
+  Typography,
+} from "@mui/material";
+import { useEffect, useState } from "react";
+import { useFormContext, useWatch } from "react-hook-form";
+import { EntityTypeEditorForm } from "./form-types";
+
+export const MultipleValuesCell = ({
+  propertyIndex,
+}: {
+  propertyIndex: number;
+}) => {
+  const { register, setValue } = useFormContext<EntityTypeEditorForm>();
+
+  const [anchorEl, setAnchorEl] = useState<HTMLDivElement | null>(null);
+  const [multipleValuesMenuOpen, setMultipleValuesMenuOpen] = useState(false);
+
+  const [array, minValue, maxValue] = useWatch({
+    name: [
+      `properties.${propertyIndex}.array`,
+      `properties.${propertyIndex}.minValue`,
+      `properties.${propertyIndex}.maxValue`,
+    ],
+  });
+
+  useEffect(() => {
+    if (minValue > maxValue) {
+      setValue(`properties.${propertyIndex}.maxValue`, minValue);
+    }
+    // Should only be triggered when minValue updates
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [propertyIndex, setValue, minValue]);
+
+  useEffect(() => {
+    if (maxValue < minValue) {
+      setValue(`properties.${propertyIndex}.minValue`, maxValue);
+    }
+    // Should only be triggered when maxValue updates
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [propertyIndex, setValue, maxValue]);
+
+  return (
+    <>
+      <TableCell
+        ref={(ref: HTMLDivElement) => setAnchorEl(ref)}
+        sx={({ palette, transitions }) => ({
+          px: "0px !important",
+          position: "relative",
+          cursor: "pointer",
+          textAlign: "center",
+          transition: transitions.create("border-color"),
+          border: `1px solid ${
+            multipleValuesMenuOpen ? palette.gray[40] : "transparent"
+          } !important`,
+          "&:hover": {
+            borderColor: `${palette.gray[40]} !important`,
+          },
+        })}
+        onClick={() => setMultipleValuesMenuOpen(true)}
+      >
+        <Box
+          sx={({ palette, transitions }) => ({
+            display: "inline-flex",
+            borderRadius: "4px 30px 30px 4px",
+            backgroundColor: "transparent",
+            transition: transitions.create(["padding", "background-color"]),
+            ...(array && !multipleValuesMenuOpen
+              ? {
+                  py: 0.5,
+                  px: 0.75,
+                  background: palette.gray[20],
+                }
+              : {}),
+          })}
+        >
+          <Checkbox
+            sx={{
+              zIndex: 1,
+            }}
+            checked={array}
```

Comment:
You shouldn't need to use `watch` for this. `register` works by passing a ref and manually updating the value when it changes to avoid component re-renders. You're breaking that performance optimisation here. Does `Checkbox` not take a ref or something? Have you looked into how to use react-hook-form with checkboxes?

Reply:
@luisbettencourt reported via Slack that checkboxes weren't reflecting a persisted checked state when using `register`. I thought I could make this work by just wrapping the component and setting `defaultChecked`, but that doesn't work when the value is set via a ref (i.e, through reset). See my aborted attempt at https://codesandbox.io/s/sparkling-dust-2fb5p9?file=/src/App.tsx

You'll just have to use the component in controlled mode, it seems there's no way to make setting the value via ref work. But you want to isolate the re-render using `Controller`, not `useController`. See a working version at https://codesandbox.io/s/sad-river-9p82tr?file=/src/App.tsx

Reply:
Thank you @nathggns, I've implemented your suggestion in 89fd36d88b1c34a1037550fafae2eb287234492b.